ui_framework/components/prompt_for_items/prompt_for_items.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import classnames from 'classnames';
+
+import { KuiButtonIcon, KuiLinkButton } from '../button';
+import { KuiPromptForItemsMessage } from  './prompt_for_items_message';
+
+export function KuiPromptForItems({ itemType, promptMessage, promptButtonText, addHref, className, ...rest }) {
+  const classes = classnames('kuiPromptForItems', className);
+  return (
+    <div className={classes} {...rest} >
+      <KuiPromptForItemsMessage>
+        {promptMessage}
+      </KuiPromptForItemsMessage>
+
+      <div className="kuiPromptForItems__actions">
+        <KuiLinkButton
+          icon={<KuiButtonIcon type="create" />}
+          aria-label={`Add a new ${itemType}`}
+          className="testClass1 testClass2"
+          data-test-subj={`addNewPromptButton`}
+          buttonType="primary"
+          href={addHref}
+        >
```

Comment:
I'm not _opposed_ to this but I do see some advantages toward not hard-coding this button in so tightly. For example, what if the interface for this component was like this:

```jsx
<KuiPromptForItems
  actions=(
    <KuiLinkButton
      icon={<KuiButtonIcon type="create" />}
      aria-label="Add a new item"
      data-test-subj="addNewItemPromptButton"
      buttonType="primary"
      href="/item/new"
    />
  )
>
  Uh oh, looks like you don't have any items yet.
</KuiPromptForItems>
```

I think the biggest benefit here is greater flexibility over those actions. You can substitute them for regular buttons if you want/have to use a click handler instead of a route change, you can change the buttonType if that makes sense for whatever reason, you can specify a different icon should that make sense, and you can specify the `data-test-subj` to make it clearer which element you're interacting with in a page object.

Lesser benefits are that an `actions` prop could accommodate an array of buttons, or links, or whatever we want should the situation call for multiple actions or different kinds of actions.

I also just like using the `children` prop where possible. 😄 

@Stacey-Gammon @weltenwort @kjbekkelund I'd love to hear your thoughts... I don't want to be dogmatic about "always compose! never encapsulate!"; I just want to pass this alternative by you.